From: Sebastian Andrzej Siewior <sebastian@breakpoint.cc>
Date: Sun, 24 Nov 2024 20:06:15 +0100
Subject: [PATCH] test: Don't check for the exact size of compressed content.

The testsuite checks for the exact size of the compressed content. The
exact size can change if the zlib library is using a different strategy.
In thise case using zlib-ng results in a slightly larger content leading
to a failure in the test.

Check that the compressed content is less than 10MiB which is a tenth of
the orignal content and proves that compression works.

Signed-off-by: Sebastian Andrzej Siewior <sebastian@breakpoint.cc>
---
 test/test.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/test/test.cc b/test/test.cc
index 33ef30b4963b..1ea28e9fe123 100644
--- a/test/test.cc
+++ b/test/test.cc
@@ -4082,7 +4082,7 @@ TEST_F(ServerTest, PutLargeFileWithGzip2) {
   ASSERT_TRUE(res);
   EXPECT_EQ(StatusCode::OK_200, res->status);
   EXPECT_EQ(LARGE_DATA, res->body);
-  EXPECT_EQ(101942u, res.get_request_header_value_u64("Content-Length"));
+  EXPECT_TRUE(res.get_request_header_value_u64("Content-Length") < (10 * 1024 * 1024));
   EXPECT_EQ("gzip", res.get_request_header_value("Content-Encoding"));
 }
 
-- 
2.45.2

